### ISSUE TITLE
Quesetioin about percent_broken in cmpass

### ISSUE THREAD

**Issue by CNS-King** [tilman151/rul-datasets#67]
Thank you very much for the help of your code. I want to ask you something I don't understand.

When I was using CMPASS data to reproduce Latent Alingn, I wanted to truncate the degraded data at 50% of the target domain, I used percent_broken=0.5, but it turned out that the data was being intercepted at 50% of the total data, not 50% of the degraded data.

In my understanding, Latent Alingn paper needs to truncate at 50% of the degraded data. Do you think my understanding is correct? If I want to degrade the data at the 50% stage, what should I do according to the rul_datsets.

Thank you very much for the help of your code, which is of great help to my research on transfer learning and remaining life prediction. I hope you can answer my questions in your spare time.

![Image](https://github.com/user-attachments/assets/307a0004-7205-4b4a-8687-74c3eda9f012)

**Comment by tilman151** [comment]
Hey, 
the reader has an argument called `truncate_degraded_only`. Setting it to `True` should do exactly what you want. Here are the [docs](https://krokotsch.eu/rul-datasets/api/rul_datasets/reader/cmapss/).

**Comment by CNS-King** [comment]
> Hey, the reader has an argument called `truncate_degraded_only`. Setting it to `True` should do exactly what you want. Here are the [docs](https://krokotsch.eu/rul-datasets/api/rul_datasets/reader/cmapss/).

Thank you for taking the time to answer my question. Your guidance has saved me significant time and effort. 

I have another question, in the code example in rul_adapt/doc/example, is there a way to view the training loss function curve after the training is completed? During training, if the epoch is several hundred, is there a way to keep the optimal model during training in order to prevent overfitting? I would like to ask if this is possible in the rul_adapt library.

![Image](https://github.com/user-attachments/assets/06deb16f-30f9-4d39-b6f3-8875b7cfbefe)

**Comment by tilman151** [comment]
The things you are asking about are not part of rul-adapt. They are part of the underlying PyTorch Lightning library.

You can pass a logger to the trainer class to see metrics and other logs. Which one is up to you. I used [the WandB logger](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.loggers.wandb.html#module-lightning.pytorch.loggers.wandb). Tensorboard is supported, too.

Keeping the best model is done by passing a checkpoint callback to the trainer. You can read about it [here](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.ModelCheckpoint.html#lightning.pytorch.callbacks.ModelCheckpoint).

**Comment by CNS-King** [comment]
I'm sorry to bother you again.
I have a question in rul_datasets LatentAlignDataModule. When loading the data using this module, why are six parts of the data returned? Through analysis, I feel that the dataset = AdaptionDataset(source_degraded, target_degraded, healthy) in the LatentAlignDataModule module returns the six batches of data. After looking at the data in the AdaptionDataset, I still don't understand what data was processed and what data was returned.

I am looking forward to your reply.

![Image](https://github.com/user-attachments/assets/c275bfbe-92f9-4331-bc1e-c5092b8be916)

![Image](https://github.com/user-attachments/assets/bb235eca-ded5-45a4-adb4-233d8b36c59f)
